admin_panel/exams/viewset.py

```diff
@@ -0,0 +1,69 @@
+# viewsets.py
+from django_filters.rest_framework import DjangoFilterBackend
+from drf_yasg.utils import swagger_auto_schema
+from rest_framework import viewsets
+from admin_panel.exams.model import Exam
+from admin_panel.exams.serializer import ExamCreateSerializer, ExamResponseSerializer, ExamSerializer
+
+
+class ExamViewSet(viewsets.ModelViewSet):
+    queryset = Exam.objects.all()
+    filter_backends = [DjangoFilterBackend]
+    filterset_fields = ['organization', 'difficulty', 'objective']
+
+    def get_serializer_class(self):
+        if self.action in ['create']:
+            return ExamCreateSerializer
+        if self.action in ['update', 'partial_update']:
+            return ExamSerializer
+        return ExamResponseSerializer
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider mapping dict for serializer selection</b></div>
<div id="fix">

Consider using a dictionary mapping for action-to-serializer instead of multiple if conditions. This would make the code more maintainable and easier to extend.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -14,6 +14,11 @@
     def get_serializer_class(self):
 -        if self.action in ['create']:
 -            return ExamCreateSerializer
 -        if self.action in ['update', 'partial_update']:
 -            return ExamSerializer
 -        return ExamResponseSerializer
 +        serializer_map = {
 +            'create': ExamCreateSerializer,
 +            'update': ExamSerializer,
 +            'partial_update': ExamSerializer
 +        }
 +        return serializer_map.get(self.action, ExamResponseSerializer)
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/sleep-404/ExamEvoke/pull/8#issuecomment-2499580485>#1f7607</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged